File: dehaze-java/src/main/java/com/pei/dehaze/service/LoginLogService.java
package com.pei.dehaze.service;

import com.baomidou.mybatisplus.core.metadata.IPage;
import com.pei.dehaze.model.query.LoginLogQuery;
import com.pei.dehaze.model.vo.LoginLogVO;

public interface LoginLogService {

    void recordLogin(Long userId, String username, String ip, int status, String message, String browser, String os, String deviceType);

    /**
     * 分页查询登录日志（MongoDB）
     *
     * @param query            查询条件（用户名/IP/状态/设备类型/时间范围精确匹配）
     * @param restrictedUserId 非空时仅查询该用户的日志（普通用户数据权限限定）
     */
    IPage<LoginLogVO> pageLoginLogs(LoginLogQuery query, Long restrictedUserId);
}

File: dehaze-go/internal/service/login_log/login_log_service.go
package login_log

import (
	"context"
	"time"

	"github.com/earthyzinc/dehaze-go/internal/model"
	"github.com/earthyzinc/dehaze-go/internal/model/query"
	loginlogrepo "github.com/earthyzinc/dehaze-go/internal/repository/login_log"
	"github.com/earthyzinc/dehaze-go/pkg/common"
	"github.com/earthyzinc/dehaze-go/pkg/logger"
	"go.uber.org/zap"
)

type LoginLogService struct {
	repo *loginlogrepo.LoginLogRepository
}

func NewLoginLogService(repo *loginlogrepo.LoginLogRepository) *LoginLogService {
	return &LoginLogService{repo: repo}
}

func (s *LoginLogService) RecordLogin(ctx context.Context, userID *int64, username, ip string, status int, message, browser, os, deviceType string) error {
	if deviceType == "" {
		deviceType = "web"
	}
	log := &model.LoginLog{
		UserID:     userID,
		Username:   username,
		IP:         ip,
		Browser:    browser,
		OS:         os,
		DeviceType: deviceType,
		Status:     status,
		Message:    message,
		CreateTime: time.Now(),
	}
	if err := s.repo.Create(ctx, log); err != nil {
		logger.Error("写入登录日志失败", zap.Error(err))
		return err
	}
	return nil
}

// PageLogs 分页查询登录日志
func (s *LoginLogService) PageLogs(ctx context.Context, q *query.LoginLogQuery) ([]model.LoginLog, int64, error) {
	logs, total, err := s.repo.PageLogs(ctx, q)
	if err != nil {
		return nil, 0, common.WrapBizError(common.DATABASE_ERROR, "查询登录日志失败", err)
	}
	return logs, total, nil
}
